integrations/kml-parser/src/main/java/ch/stayhome/integrations/kml/job/ContactInformationGuesser.java

```diff
@@ -0,0 +1,67 @@
+package ch.stayhome.integrations.kml.job;
+
+import static org.apache.commons.lang3.StringUtils.trim;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.i18n.phonenumbers.PhoneNumberMatch;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+import org.apache.commons.lang3.StringUtils;
+
+
+public class ContactInformationGuesser {
+
+	private final PhoneNumberUtil util = PhoneNumberUtil.getInstance();
+
+	public String extractPhoneNumber(String text) {
+		text = StringUtils.chomp(text);
+		final Iterable<PhoneNumberMatch> numbers = util.findNumbers(text, "CH");
+		final List<String> results = new ArrayList<>();
+		numbers.forEach(phoneNumberMatch -> {
+			final Phonenumber.PhoneNumber number = phoneNumberMatch.number();
+			results.add(util.format(number, PhoneNumberUtil.PhoneNumberFormat.INTERNATIONAL));
+		});
+		return results.isEmpty() ? "" : trim(results.get(0));
+	}
+
+	public String extractEmail(String text) {
+		text = StringUtils.chomp(text);
+		Matcher m = Pattern.compile("[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+").matcher(text);
```

Comment:
This is a rather restrictive pattern to match email addresses since the local part of an address allows pretty much any ascii character. Though everything besides a-z0-9. is rare, "!#$%&'*+-/=?^_`.{|}~@example.com" is perfectly valid. With restrictions, even white space is allowed :-D

Reply:
As we have discussed: I'll keep the current regex for now since 100% of the data that I crawled machted -> later we might need to adapt it